pkg/jobs/structured_log.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2024 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package jobs
+
+import (
+	"context"
+
+	"github.com/cockroachdb/cockroach/pkg/jobs/jobspb"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
+	"github.com/cockroachdb/redact"
+)
+
+type ProgressHighWatermark struct {
+	WallTime int64 `json:"wall_time"`
+	Logical  int32 `json:"logical"`
+}
+
+type ProgressFractional struct {
+	FractionCompleted float32 `json:"fraction_completed;omitempty"`
+}
+
+type JobStateChange struct {
+	JobID   jobspb.JobID `json:"job_id"`
+	JobType string       `json:"job_type"`
+	Status  Status       `json:"status"`
+	// TODO(abarganier): we need to consider redaction for this string field.
+	RunningStatus        redact.RedactableString `json:"running_status,omitempty"`
+	StateChangeTimeNanos int64                   `json:"state_change_time_nanos"`
+	ModifiedMicros       int64                   `json:"modified_micros"`
+	// TODO(abarganier): Figure out redactable errors.
+	// TODO(abarganier): Payload struct has error, resume_errors, cleanup_errors, final_resume_error,
+	// do we need to include these?
```

Comment:
^this is sad lol. It looks like `ResumeErrors`, and `CleanupErrors` aren't actually used and we, the job maintaners, should delete these fields.

For now, lets include FinalResumeError and Error if they are not nil?